glue/sales_val_flagging.py

```diff
@@ -152,6 +149,63 @@ def group_size_adjustment(df, stat_groups: list, min_threshold, condos: bool):
         df: dataframe with newly manually adjusted outlier values.
 
     """
+
+    # Assign sv_outlier_reasons
+    df["sv_outlier_reason1"] = "Null"
+    df["sv_outlier_reason2"] = "Null"
+    df["sv_outlier_reason3"] = "Null"
+
+    cols_to_iterate = [col for col in df.columns if col.startswith("sv_ind")]
+
+    def fill_outlier_reasons(row):
+        reasons = [
+            row[f"sv_outlier_reason{i}"] for i in range(1, 4)
+        ]  # Collect current reasons
+
+        for col in cols_to_iterate:
+            if row[col]:  # If the condition is True
+                reason = col  # Use column name as reason
+                # Check if reason is not already recorded and there's a 'Null' slot to fill
+                if reason not in reasons and "Null" in reasons:
+                    next_index = reasons.index("Null") + 1  # Find the next 'Null' index
+                    row[f"sv_outlier_reason{next_index}"] = reason
+                    reasons[next_index - 1] = reason  # Update reasons list
+
+        return row
+
+    df = df.apply(fill_outlier_reasons, axis=1)
+
+    outlier_type_crosswalk = {
+        "sv_ind_ptax_flag_w_deviation": "PTAX-203 Exclusion",
+        "sv_ind_price_high_price": "High price",
+        "sv_ind_price_low_price": "Low price",
+        "sv_ind_price_high_price_sqft": "High price per square foot",
+        "sv_ind_price_low_price_sqft": "Low price per square foot",
+        "sv_ind_char_short_term_owner": "Short-term owner",
+        "sv_ind_char_family_sale": "Family Sale",
+        "sv_ind_char_non_person_sale": "Non-person sale",
+        "sv_ind_char_statistical_anomaly": "Statistical Anomaly",
+        "sv_ind_char_price_swing_homeflip": "Price swing / Home flip",
+    }
```

Comment:
Translate indicator cols to more readable / design doc specification.

Reply:
**[Suggestion, non-blocking]** If we're already creating a crosswalk data structure for all of the reason codes, I think we should use it instead of `cols_to_iterate` as the canonical list of reason codes for forward-filling. That way we can be explicit about the intended order of priority for the columns, instead of relying on the implicit order of the columns in the dataframe to establish priority (dicts are ordered starting in Python 3.7, which is the same version used by [Glue 2.0](https://docs.aws.amazon.com/glue/latest/dg/glue-version-support-policy.html)).

Reply:
I now utilize this `outlier_type_crosswalk` for the operations in this function, and delete the previous order handling in `ptax_adjustment`. Less redundancy and a defined order rather than relying on column order.